src/tools/concurrency/KomposTrace.java

```diff
@@ -169,15 +171,27 @@ boolean swapStorage(final Activity current) {
               Implementation.IMPL_CURRENT_ACTIVITY.getSize())) {
         return false;
       }
+
       super.swapStorage();
       this.lastActivity = null;
       recordCurrentActivity(current);
+
+      if (Thread.currentThread() instanceof ActorProcessingThread) {
+        EventualMessage msg = EventualMessage.getCurrentExecutingMessage();
+        ActorProcessingThread apt =
+            (ActorProcessingThread) ActorProcessingThread.currentThread();
+
+        if (apt.getCurrentActor() != null && msg != null) {
+          scopeStart(DynamicScopeType.TURN, msg.getMessageId(), msg.getTargetSourceSection());
+        }
+      }
+
       return true;
     }
 
     @TruffleBoundary
     protected boolean ensureSufficientSpace(final int requiredSpace, final Activity current) {
-      if (position + requiredSpace < VmSettings.BUFFER_SIZE) {
+      if ((position + requiredSpace) >= VmSettings.BUFFER_SIZE) {
```

Comment:
Ugh, I was constantly swapping buffers? *sigh*

Reply:
Yes, completely forgot about this bug